keps/sig-cloud-provider/003-testgrid-e2e.md

```diff
@@ -0,0 +1,224 @@
+Run: ['gsutil', '-q', '-h', 'Content-Type:text/plain', 'cp', '/Users/AndrewSyKim/go/src/k8s.io/kubernetes/results/plugins/e2e/results/e2e.log', 'gs://k8s-conformance-demo/cloud-provider-demo/e2e-conformance-release-v1.10/1528333637/artifacts/e2e.log']
+Done.
+```
+
+##### Testgrid Configuration
+
+Next thing you want to do is configure testgrid to read results from your GCS bucket. There are two [configuration](https://github.com/kubernetes/test-infra/tree/master/testgrid#configuration) steps required. One for your [test group](https://github.com/kubernetes/test-infra/tree/master/testgrid#test-groups) and one for your [dashboard](https://github.com/kubernetes/test-infra/tree/master/testgrid#dashboards).
+
+To add a test group update [config.yaml](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) with something like the following:
+```
+test_groups:
+...
+...
+- name: cloud-provider-demo-e2e-conformance-release-v1.10
+  gcs_prefix: k8s-conformance-demo/cloud-provider-demo/e2e-conformance-release-v1.10
+```
+
+To add a link to your results in the testgrid dashboard, update [config.yaml](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) with something like the following:
+```
+dashboards:
+...
+...
+- name: conformance-demo-cloud-provider
+  dashboard_tab:
+  - name: "Demo Cloud Provider, v1.10"
+    description: "Runs conformance tests for cloud provier demo on release v1.10"
+    test_group_name: cloud-provider-demo-e2e-conformance-release-v1.10
+```
+
+Once you've made the following changes, open a PR against the test-infra repo. Once your PR merges you should be able to view your results on https://k8s-testgrid.appspot.com/ which should be ready to be consumed by the necessary stakeholders (sig-release, sig-testing, etc).
+
+
+#### Examples
+
+Here are some more concrete examples of how other cloud providers are running conformance tests and uploading results to testgrid:
+* Open Stack
+    * [OpenLan zuul job for running/uploading testgrid results](https://github.com/theopenlab/openlab-zuul-jobs/tree/master/playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance)
+    * [OpenStack testgrid config](https://github.com/kubernetes/test-infra/pull/7670)
+    * [OpenStack conformance tests dashboard](https://github.com/kubernetes/test-infra/pull/8154)
+
+
+### Risks and Mitigations
+
+#### Operational Overhead
+
+Operating CI/CD system to run conformance tests on a regular basis may incur extra work from every cloud provider. Though we anticipate the benefits of running conformance tests to outweight the operational overhead, in some cases it may not.
+
+Mitigation: TODO
+
+#### Misconfigured Tests
+
+There are various scenarios where cloud providers may mistakenly upload incorrect conformance tests results. One example being uploading results for the wrong Kubernetes version.
+
+Mitigation: TODO
```

Comment:
@AishSundar @BenTheElder would love some feedback on how to mitigate this going forward

Reply:
same with flaky tests below

Reply:
If the providers own their own GCS bucket, then they can go into the appropriate folder and delete data corresponsing to the run / move it to the right directory in the bucket etc. If the GCS bucket is owned by GKE EngProd, then they can mail gke-kubernetes-engprod@google.com, with details of the run and bucket, for fixing the data. 

@BenTheElder wdyt? accurate?

Reply:
So TestGrid processes X results back until it stops finding changes, but that means that it varies how many columns back it looks. If you remove a GCS entry and want that run to not show up in TestGrid, the best way to enforce that is to force TestGrid to re-collect results for that test group.

To force TestGrid to re-collect, you can:
- Increase the test group's `days_of_results` field, [Recommended]
- Change the test group's `query` or `column_header` fields, or
- Change the test group name (which TestGrid will treat as a new test group).

If you need to wipe something urgently, you can also request that I or someone else on the team delete the state file for the test group, which will remove the results until TestGrid updates that group again.

Reply:
Small note: the changes mentioned above would be to the testgrid [config file](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.yaml) in test-infra, by a PR.
thanks @michelle192837 :-)

Reply:
This is really good to know, I'll add a section about this.